### ISSUE TITLE
After GUI Error Message keyboards events ignored?

### ISSUE THREAD

**Issue by Zeroes1** [microsoft/terminal#15956]
### Windows Terminal version

1.18.1462.0

### Windows build number

10.0.19045.3324

### Steps to reproduce

portable
admin
default settings WT

trouble:

under CMD/FAR Manager

Try run command:

like:
start 123

where 123 - not existed file in %path%

Windows show _GUI_ error message: can't find 123. Check name and try again...                     
close message by Enter/Esc/AltF4 or mouse click on button OK
we return to WT and see what in console show message too: can't find file 123

AND now ... drumroll ...

### Expected Behavior

keyboard events work, i press any key and see results

### Actual Behavior

keybord not work, press any keys but nothing print... (keyboards events ignored?)
treatment method: click by mouse (any keys) inside WT window

the problem is 100% reproducible

**Comment by Zeroes1** [comment]
under 
Microsoft.WindowsTerminal_1.17.11461.0_x64.zip
same behavior

**Comment by codecruisedor** [comment]
Yes, I can see the same issue in Terminal Version 1.17.11461.0. Seems like after the message is closed by clicking OK or clicking the close button, the focus is not returned to the WT rather we have to once click on it for the focus to return.

**Comment by Zeroes1** [comment]
Hmm, where is the focus? :) i alt-tab and alt-tab again (return to WT window) -not help...

**Comment by zadjii-msft** [comment]
I'm 99% sure I know what's going on here. 

The `MessageBox` is parented to the console HWND. But for the Terminal, that HWND is not the Terminal's HWND - it's a hidden "pseudo hwnd". So when the MessageBox gets dismissed, and it returns foreground to its parent - the window that gets foreground is NOT the Terminal's window. 

This is also probably **H**ard to solve. I'm not sure we get notified when we get foreground. And #13388 has some details on why yeeting foreground to the Terminal is hard.

**Comment by Zeroes1** [comment]
may be interest for someones how to workaround this trouble...
for example if use AutoHotkey v2.x [https://www.autohotkey.com/]

```
;------------------------- Restore Focus Windows Terminal if Lost -----------------------------
SetTimer LostFocusWT, 1000

LostFocusWT()
{
 Try
 {
  if (WinGetClass("A") == "PseudoConsoleWindow")
   {
      if WinExist("ahk_class CASCADIA_HOSTING_WINDOW_CLASS")
      WinActivate
   }
 }
}
```

**Comment by Zeroes1** [comment]
@zadjii-msft thanks it's work!
